packages/synapse-react-client/src/lib/utils/hooks/useDetectSSOCode.ts

```diff
@@ -53,26 +52,17 @@ export default function useDetectSSOCode(
     // state is used during OAuth based Synapse account creation (it's the username)
     if (code && provider) {
       const redirectUrl = `${redirectURL}?provider=${provider}`
-      const redirectAfterSuccess = () => {
-        // go back to original route after successful SSO login
-        const originalUrl = localStorage.getItem('after-sso-login-url')
-        localStorage.removeItem('after-sso-login-url')
-        if (originalUrl) {
-          window.location.replace(originalUrl)
-        }
-      }
```

Comment:
This redirect logic seems to me like it should be app-specific, so it's now an argument `onSignInComplete`